### ISSUE TITLE
Make the history for daily hourly max usage persistent across months (currently the stats are dropped at new months)

### ISSUE THREAD

**Issue by frodeheg** [frodeheg/no.sparegris#43]
Deadline: a few days before the end of the month so the already gathered statistics so far are not lost.

Proposed archive solution:
```
/*
* The archive is indexed as follows:
 *   archive[dataId][period][time][data]
 * Where the parameters are:
 *   dataId:
 *     - "maxPower"           : value : The maximum power measured in the interval (for monthly this is the 3-day average)
 *     - "dataOk"             : bool  : Indicating if the data is reliable or if a reboot / app restart was detected
 *     - "powUsage"           : value : The measured power usage for the given interval
 *     - "moneySavedTariff"   : value : The money saved by upholding the tariff in the interval
 *     - "moneySavedUsage"    : value : The money saved by moving power in the interval
 *     - "price"              : value : The electricity price per hour, for period day/month/year = averages
 *     - "pricePoints"        : array : The price point per hour, for period day/month/year then array [0..4] with number of occurances of the price point
 *   period:
 *     - "yearly" : One item stored per year (never expires)
 *     - "monthly": One item stored per month (never expires)
 *     - "daily"  : One item stored per day (expires after ARCHIVE_EXPIRE_TIME_DAILY)
 *     - "hourly" : One item stored per hour (expires after ARCHIVE_EXPIRE_TIME_HOURLY)
 *   time:
 *     - "YYYY" for period "yearly"
 *     - "YYYY" for period "monthly"
 *     - "YYYY-MM" for period "daily"
 *     - "YYYY-MM-DD" for period "hourly"
 *   data:
 *     - Single value for period "yearly"
 *     - Array going from 0 to 11, one value for every month for period "monthly"
 *     - Array going from 0 to monthDays-1, one value for every day in the month for period "daily"
 *     - Array going from 0 to 23, one value for every hour for period "hourly"
  */
```

**Comment by frodeheg** [comment]
I will make graphs of the tracked data later.
If anyone want other graphs than for the data suggested above, please make a note

What should the expiration time be set to?
* 2 months ago?
* 1 year ago?
(note that it's only the daily data that will expire, and I will make an exporter)

**Comment by frodeheg** [comment]
The archive system has been implemented in version 0.18.25, however it is at current only a redundant system on top of the old system. Making use of the new archive system has been branched out into #54.
